### PR TITLE
Added DRAW_ROUND response in onDrawTile lambda

### PR DESCRIPTION
**Pull request by jzc1828**
- [x] Add `DRAW_ROUND` response in `onDrawTile lambda`
- [x] Refactor `DRAW_TILE` related logic to send empty string when reached the end of the mahjong wall
(found a bug which skips an index every time user draws a tile. Should fix it with this patch)
- [x] Modify `DrawTilePayload` as
```ts
export interface DrawTilePayload {
  tile: string;
  currentIndex: number;
}
```
- [x] Add `DrawRoundPayload` as
```ts
export interface DrawRoundPayload {
  gameId: string;
  connectionId: string;
}

```
- [x] More manual testing to make sure

1. Tile can be drawn successfully and wall index should be incremented accordingly
2. When index reaches to the end of the wall (index 144), `drawTile` should return an empty string from the DB

### CODE REVIEW

File: src/dynamodb/gameStateDBService.ts [hunk 1]
```diff
@@ -190,16 +190,17 @@ export const incrementCurrentTileIndex = async (gameId: string): Promise<GameSta
...
 
-  // Draw a tile and increment index by 1
-  if (currentGameState) {
-    const { wall, currentIndex } = currentGameState;
-    await incrementCurrentTileIndex(gameId);
-    tileDrawn = wall[currentIndex];
+  // Return empty string if index reach 143
+  if (currentIndex === 143) {
+    return tileDrawn;
   }
```
Comment (yummyblabla) [inline src/dynamodb/gameStateDBService.ts:199]: How about
```suggestion
  if (currentIndex >= wall.length) {
    return tileDrawn;
  }
```

File: src/websocket/broadcast/gameBroadcast.ts [hunk 2]
```diff
@@ -228,24 +230,58 @@ export const broadcastGameStart = async (
...
+    await broadcastDrawRound(ws, gameId, connectionId, connectionIds);
+    return;
+  }
 
-  return tileDrawn;
+  console.log(`Tile ${tileDrawn} was drawn by ${connectionId}`);
+  const wsResponse = createDrawTileResponse({
+    tile: tileDrawn,
+    currentIndex: (await getCurrentTileIndex(gameId)) as number,
+  });
```
Comment (yummyblabla) [inline src/websocket/broadcast/gameBroadcast.ts:283]: I see you're passing in currentIndex as well. I will update front-end for this change

File: src/websocket/broadcast/gameBroadcast.ts [hunk 3]
```diff
@@ -228,24 +230,58 @@ export const broadcastGameStart = async (
...
+    }
+
+    // Users must exist in the game
+    const users = await getUsersInGame(gameId);
+    if (!users) throw Error('broadcastDrawTileToUser: No user found in game, failed to broadcast DRAW_ROUND');
+
+    const connectionIds = getConnectionIdsFromUsers(users);
+    await broadcastDrawRound(ws, gameId, connectionId, connectionIds);
+    return;
+  }
```
Comment (yummyblabla) [inline src/websocket/broadcast/gameBroadcast.ts:277]: Looks good

File: src/websocket/broadcast/gameBroadcast.ts (hunk 2 above)
Comment (jzc1828) [reply src/websocket/broadcast/gameBroadcast.ts:283]: kk

File: src/dynamodb/gameStateDBService.ts (hunk 1 above)
Comment (jzc1828) [reply src/dynamodb/gameStateDBService.ts:199]: kk

File: src/websocket/broadcast/gameBroadcast.ts (hunk 3 above)
Comment (jzc1828) [reply src/websocket/broadcast/gameBroadcast.ts:277]: thx

**Pull request merged**